src/gui/filedetails/sharemodel.cpp

```diff
@@ -253,7 +253,9 @@ void ShareModel::updateData()
     qCDebug(lcShareModel) << "Updating share model data now.";
 
     const auto relPath = _localPath.mid(_folder->cleanPath().length() + 1);
-    _sharePath = _folder->remotePathTrailingSlash() + relPath;
+    const auto sharePath = _folder->remotePathTrailingSlash() + relPath;
+    const auto percentEncoded = QUrl::toPercentEncoding(sharePath);
+    _sharePath = QString::fromUtf8(percentEncoded);
```

Comment:
I would probably move this into a private utility method such that it is easier to grasp what is going on

Reply:
Done